pkg/migrate/command.go

```diff
@@ -0,0 +1,138 @@
+package migrate
+
+import (
+	"database/sql"
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/golang-migrate/migrate"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/skygeario/skygear-server/pkg/migrate/database/postgres"
+)
+
+func Run(module string, schema string, databaseURL string, sourceURL string, dryRun bool, commandArgs []string) (err error) {
+	if schema == "" {
+		err = errors.New("missing schema")
+		return
+	}
+
+	if module == "" {
+		err = errors.New("missing module")
+		return
+	}
+
+	if databaseURL == "" {
+		err = errors.New("missing db url")
+		return
+	}
+
+	if sourceURL == "" {
+		err = errors.New("missing source url")
+		return
+	}
+
+	versionTable := fmt.Sprintf("_%s_version", module)
+
+	db, err := sql.Open("postgres", databaseURL)
+	if err != nil {
+		return
+	}
+
+	_, err = db.Exec(fmt.Sprintf("SET search_path TO %s", schema))
+	if err != nil {
+		return
+	}
+
+	config := postgres.Config{
+		MigrationsTable: versionTable,
+		DryRun:          dryRun,
+	}
+	driver, err := postgres.WithInstance(db, &config)
+	if err != nil {
+		return
+	}
+	defer driver.Close()
+
+	m, err := migrate.NewWithDatabaseInstance(sourceURL, "postgres", driver)
+	if err != nil {
+		return
+	}
+
+	var arg0, arg1 string
+	if len(commandArgs) > 1 {
+		arg1 = commandArgs[1]
+	}
+	if len(commandArgs) > 0 {
+		arg0 = commandArgs[0]
+	}
+
+	err = runCommand(m, arg0, arg1)
+	return
+}
+
+func runCommand(m *migrate.Migrate, arg0 string, arg1 string) (err error) {
```

Comment:
Consider using https://github.com/spf13/pflag ?

Reply:
The golang build in `flag` has function to get the arguments by index, so we may not need this lib. I updated the function signature, see it looks better? My idea is the function doesn't need to know how the command get the arguments, and the function will not access the flag directly.

Reply:
The build in `flag` also support help like this.

```
$ go run cmd/migrate/main.go -help
Usage of /var/folders/pj/hkvxpm2123x7367kd5brbr_80000gn/T/go-build810654351/b001/exe/main:
  -database string
    	migration db url (default "postgres://postgres:@localhost/postgres?sslmode=disable")
  -dir string
    	(optional) directory of revisions files (default "cmd/migrate/revisions")
  -module string
    	module name, e.g. gateway, core, auth, record
  -schema string
    	migration schema (default "app__")
exit status 2
```

Or we think `pflag` format looks better?:)

Reply:
No, obviously I use pflag before. It looks fine. Seen we don't need to change 👍